docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,296 @@
+/**
+@page guide_developer_styleguide Style Guide
+@brief Guide for the coding style used in this SDK.
+
+The goal of this style guide is to encourage a readable and consistent coding style across the entire SDK.
+
+@section guide_developer_styleguide_codingstyle Coding Style
+@brief The coding style used in this SDK.
+
+The coding style aims to produce code that is readable and easy to debug. An example is provided in @ref guide_developer_styleguide_codingstyle_example.
+
+@subsection guide_developer_styleguide_codingstyle_generalguidelines General guidelines
+@brief General guidelines for library style.
+- All new libraries (being developed in the development branch) should only use [ANSI C](https://en.wikipedia.org/wiki/ANSI_C#C89) features and standard library functions. (All libraries in the v4_beta branch used the C99 standard, so any changes to the branch can continue using C99).
+- Libraries should log extensively.
```

Comment:
Didn't we decide on a caveat to this? Something about not needing to log for pure functions? @dan4thewin

Reply:
Updated to say: ```Libraries should log extensively except in the case of pure functions.```